src/main/java/com/google/firebase/remoteconfig/Version.java

```diff
@@ -0,0 +1,225 @@
+
+import com.google.common.base.Strings;
+import com.google.firebase.internal.NonNull;
+import com.google.firebase.internal.Nullable;
+import com.google.firebase.remoteconfig.internal.TemplateResponse;
+import com.google.firebase.remoteconfig.internal.TemplateResponse.VersionResponse;
+
+import java.text.ParseException;
+import java.util.Objects;
+
+/**
+ * Represents a Remote Config template version.
+ * Output only, except for the version description. Contains metadata about a particular
+ * version of the Remote Config template. All fields are set at the time the specified Remote
+ * Config template is published. A version's description field may be specified when
+ * publishing a template.
+ */
+public final class Version {
+
+  private final String versionNumber;
+  private final long updateTime;
+  private final String updateOrigin;
+  private final String updateType;
+  private final User updateUser;
+  private final String rollbackSource;
+  private final boolean legacy;
+  private String description;
+
+  private Version() {
+    this.versionNumber = null;
+    this.updateTime = 0L;
+    this.updateOrigin = null;
+    this.updateType = null;
+    this.updateUser = null;
+    this.rollbackSource = null;
+    this.legacy = false;
+  }
+
+  Version(@NonNull VersionResponse versionResponse) {
+    checkNotNull(versionResponse);
+    this.versionNumber = versionResponse.getVersionNumber();
+
+    if (!Strings.isNullOrEmpty(versionResponse.getUpdateTime())) {
+      try {
+        this.updateTime = RemoteConfigUtil.convertToMilliseconds(versionResponse.getUpdateTime());
+      } catch (ParseException e) {
+        throw new IllegalStateException("Unable to parse update time.", e);
+      }
+    } else {
+      this.updateTime = 0L;
+    }
+
+    this.updateOrigin = versionResponse.getUpdateOrigin();
+    this.updateType = versionResponse.getUpdateType();
+    TemplateResponse.UserResponse userResponse = versionResponse.getUpdateUser();
+    this.updateUser = (userResponse != null) ? new User(userResponse) : null;
+    this.description = versionResponse.getDescription();
+    this.rollbackSource = versionResponse.getRollbackSource();
+    this.legacy = versionResponse.isLegacy();
+  }
+
+  /**
+   * Creates a new {@link Version} with a description.
+   */
+  public static Version withDescription(String description) {
+    return new Version().setDescription(description);
+  }
+
+  /**
+   * Gets the version number of the template.
+   *
+   * @return The version number or null.
+   */
+  @Nullable
+  public String getVersionNumber() {
+    return versionNumber;
+  }
+
+  /**
+   * Gets the update time of the version. The timestamp of when this version of the Remote Config
+   * template was written to the Remote Config backend.
+   *
+   * @return The update time of the version or null.
+   */
+  @Nullable
+  public long getUpdateTime() {
+    return updateTime;
+  }
+
+  /**
+   * Gets the origin of the template update action.
```

Comment:
This means console or backend, correct?  Might be worth noting.

Reply:
I think it could be the Console, Admin Node, Admin Java, and the CLI. This could change in the future as we add more SDKs. Should we mention all that here?

Reply:
I think you're right -- let's not go down that path.